dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/imports/validation/EnrollmentSecurityImportValidationTest.java

```diff
@@ -280,4 +290,88 @@ void testUserHasNoAccessToProgramTeiType() throws IOException {
 
     assertHasOnlyErrors(importReport, ValidationCode.E1104);
   }
+
+  @Test
+  void shouldFailWhenTeNotEnrolledAndUserHasNoAccessToTeRegisteringOrgUnit() throws IOException {
+    clearSecurityContext();
```

Comment:
Can you think of a way to make this test setup more readable? Its really hard to see what the reason is why this test will fail the import and why the other one will make it pass 😅